woocommerce/woocommerce
# Fix product attributes lookup data and filter by attribute widget with non-ASCII named attributes

### All Submissions:






### Changes proposed in this Pull Request:



This pull request fixes three bugs related to the product attributes lookup table that were apparent when using an attribute with non-ASCII characters for variations:

1. In PHP 8, saving a variable product that uses a non-ASCII attribute for variations throws a fatal error.
2. An entry is generated in the lookup table for each term of the variation, not just for those actually used to define variations.
3. Also the "Filter product by attribute" widget doesn't display.

The lookup table will require regeneration, at least for the affected products, after applying this fix.
   
### How to test the changes in this Pull Request:

1. Start with PHP 8 and without the fix.
2. Make sure that you have product attributes lookup table enabled, including the option for direct updates (Settings - Products - Advanced).

![image](https://user-images.githubusercontent.com/937723/186134938-c54b3db3-8182-45f4-b145-c69428393633.png)

5. Create a global product attribute named 大きさ with these three terms: 中 大 小
6. Add a "Filter by attribute" widget for the above in the shop.
7. Create a variable product that uses that attribute for variations (with all three terms), but create only one variation for 大
8. Save the product, you'll get a fatal error.
9. Switch to PHP 7 and repeat the above.
10. Create a second variable product that uses that attribute for variations, this time with only one variation for 小
11. Go to the shop page, the filter by attribute widget is missing.
12. Apply the fix.
13. Reload the shop page, the widget is now visible but it displays incorrect information (as if the variations had been defined for "Any" value of the attribute):

![FilterByJapaneseSize1](https://user-images.githubusercontent.com/937723/186135764-3c32d2c4-9fdc-433f-8637-cf9a76a1b572.png)

14. Go to Status - Tools and run "Regenerate the product attributes lookup table", either for the entire table or for each of the products you just created.
15. Reload the shop page, this time the widget shows correct information:

![FilterByJapaneseSize2](https://user-images.githubusercontent.com/937723/186136107-ab4a1bb4-e4dc-4645-b9bd-cad4cc9771c2.png)

### Other information:




### FOR PR REVIEWER ONLY:
